src/compile/scale.ts

```diff
@@ -41,35 +41,132 @@ export type ScaleComponents = {
   binColorLegend?: ScaleComponent
 }
 
-export function initScale(topLevelSize: number, mark: Mark, channel: Channel, fieldDef: ChannelDefWithScale, scaleConfig: ScaleConfig): Scale {
-  let scale: Scale = duplicate((fieldDef || {}).scale || {});
+export function channelScalePropertyIncompatability(channel: Channel, propName: string): string {
```

Comment:
This function should return `{compatible: bool, message: string}` instead of undefined for valid. See https://basarat.gitbooks.io/typescript/content/docs/tips/null.html